File: playground/dart/bin/insights.dart
import 'package:algoliasearch/algoliasearch.dart';
import 'package:dotenv/dotenv.dart';

void main() async {
  final dotenv = DotEnv()..load(['../.env']);

  // Creating an instance of the Insights client with the provided App ID and API key.
  final insights = InsightsClient(
    appId: dotenv['ALGOLIA_APPLICATION_ID']!,
    apiKey: dotenv['ALGOLIA_SEARCH_KEY']!,
    options: ClientOptions(logger: print),
  );
  final indexName = dotenv['SEARCH_INDEX']!;

  // Creating an InsightsEvents object with a list of EventsItems objects.
  // Each EventsItems represents an event such as a user viewing a specific item.
  final events = InsightsEvents(
    events: [
      ViewedObjectIDs(
        eventType: ViewEvent.view,
        eventName: 'View event',
        index: indexName,
        userToken: 'anonymous',
        objectIDs: ['5477500'],
      )
    ],
  );
  await insights.pushEvents(insightsEvents: events);

  // Close the client and dispose of all underlying resources.
  insights.dispose();
}

File: playground/go/insights.go
package main

import (
	"fmt"

	"github.com/algolia/algoliasearch-client-go/v4/algolia/insights"
)

func testInsights(appID, apiKey string) int {
	insightsClient := insights.NewClient(appID, apiKey, insights.US)

	events := insights.NewInsightsEvents([]insights.EventsItems{
		insights.ClickedObjectIDsAsEventsItems(insights.NewClickedObjectIDs("myEvent",
			insights.CLICKEVENT_CLICK,
			"test_index",
			[]string{"myObjectID"},
			"myToken",
			insights.WithClickedObjectIDsTimestamp(1234567890))),
	})
	eventsResponse, err := insightsClient.PushEvents(
		insightsClient.NewApiPushEventsRequest(events),
	)
	if err != nil {
		fmt.Printf("request error with PushEvents: %v\n", err)
		return 1
	}

	printResponse(eventsResponse)

	return 0
}
